src/backend/distributed/commands/dependencies.c

```diff
@@ -230,6 +260,9 @@ ReplicateAllDependenciesToNode(const char *nodeName, int nodePort)
 		return;
 	}
 
+	/* since we are executing ddl commands lets disable propagation, primarily for mx */
+	ddlCommands = list_concat(list_make1(DISABLE_DDL_PROPAGATION), ddlCommands);
+
```

Comment:
`ExecuteCriticalRemoteCommandList` does not seem to run things in a transactino block, so this will not work correctly through a pgbouncer. Maybe use `SendCommandListToWorkerInSingleTransaction`.

Reply:
Guess the same is true for `EnsureDependenciesExistsOnAllNodes`
It refactored the way we created schema's, but that happened in a single statement.

For our testing track we might want to consider a test suite with pg_bouncer to catch these subtleties.